src/btouch.cs

```diff
@@ -648,3 +630,37 @@ static string GetWorkDir ()
 		}
 	}
 }
+
+static class ReferenceFixer
+{
+	public static void FixSDKReferences (string sdkRoot, string sdk_offset, List<string> references) => FixSDKReferences (Path.Combine (sdkRoot, sdk_offset), references);
+
+	public static void FixSDKReferences (string sdk_path, List<string> references, bool forceSystemDrawing = false)
+	{
+		FixRelativeReferences (sdk_path, references);
+		AddMissingRequiredReferences (sdk_path, references, forceSystemDrawing);
+	}
+
+	static bool ContainsReference (List<string> references, string name) => references.Any (v => Path.GetFileNameWithoutExtension (v) == name);
+	static void AddSDKReference (List<string> references, string sdk_path, string name) => references.Add (Path.Combine (sdk_path, name));
+
+	static void AddMissingRequiredReferences (string sdk_path, List<string> references, bool forceSystemDrawing = false)
+	{
+		foreach (var requiredLibrary in new string [] { "System", "mscorlib", "System.Core" }) {
+			if (!ContainsReference (references, requiredLibrary))
+				AddSDKReference (references, sdk_path, requiredLibrary + ".dll");
+		}
+		if (forceSystemDrawing && !ContainsReference (references, "System.Drawing"))
+			AddSDKReference (references, sdk_path, "System.Drawing.dll");
+	}
+
+	static bool ExistsInSDK (string sdk_path, string name) => File.Exists (Path.Combine (sdk_path, name));
+
+	static void FixRelativeReferences (string sdk_path, List<string> references)
+	{
+		foreach (var r in references.Where (x => ExistsInSDK (sdk_path, x + ".dll")).ToList ()) {
```

Comment:
Why the `.ToList ()`? It looks redundant (and slower).

Reply:
I don't believe you can remove items from a list while enumerating it. Thus we make a copy of the list to enumerate while mutating the original.

Reply:
Confirmed, check https://dotnetfiddle.net/tVGT50 and compare with and without the define commented out.

```[System.InvalidOperationException: Collection was modified; enumeration operation may not execute.]```

Reply:
OK, I didn't see that you were modifying the list.

Reply:
I don't love the pattern, but i wasn't sure of a cuter way to do it.